### PR TITLE
Switch between drawing categories

### LINKED ISSUE #168: Allow switching between drawing categories
This is a follow up to #152

Right now we only ever show one type of drawings but we have so many nice ones now.

Can you add a custom sidebar option to allow the user to switch between different drawing categories/types similar to the switching between cell types in the microscopy example?

### PR DESCRIPTION
**Pull request by eeeeenchanted**
## Description

> What was changed in this pull request?

Allow switching between drawing categories like the microscopy example.

> Why is it necessary?

Fixes #168 

## Checklist

- [x] GitHub labels properly set (e.g., bug, new feature, etc.)
- [ ] Documentation added or updated
- [ ] Examples added or updated
- [ ] Screenshot or GIF included (e.g. new or changed visualization features)
- [ ] CHANGELOG.md updated

### CODE REVIEW

File: examples/drawings.js [hunk 1]
```diff
@@ -32,6 +32,24 @@ const regionToColor = region => {
...
 };
 
+const selectedCategory = 'teapot';
+const categories = [
+  'flower',
+  'piano',
+  'power outlet',
+  'teapot',
+  'The Mona Lisa',
+  'The Great Wall of China',
```
Comment (flekschas) [inline examples/drawings.js]: Let's not include those as there wasn't much to see. If you want to keep the great wall in for sentimental reasons please do so :)

File: examples/drawings.js [hunk 2]
```diff
@@ -32,6 +32,24 @@ const regionToColor = region => {
   }
 };
 
+const selectedCategory = 'teapot';
+const categories = [
+  'flower',
+  'piano',
+  'power outlet',
```
Comment (flekschas) [inline examples/drawings.js]: Please make all names lower case and replace white spaces with dashes. Thanks
```suggestion
  'power-outlet',
```

**Comment by flekschas** [comment]
Any idea why the spinner doesn't appear when we update the items? Is it because piling.js assumes we're updating the texture?

**Comment by eeeeenchanted** [comment]
> Any idea why the spinner doesn't appear when we update the items? Is it because piling.js assumes we're updating the texture?

Yeah I think you're right. In the microscopy example since we have custom-id for items, when changing the subset the library thinks we are creating new items. But in this example, the library just thinks we're updating the texture.

I just added halt&resume method in `updateItemTexture()`. Is that ok?

File: examples/drawings.js (hunk 2 above)
Comment (eeeeenchanted) [reply examples/drawings.js]: Remember to rename the json file as well! 😄

**Comment by flekschas** [comment]
> I just added halt&resume method in updateItemTexture(). Is that ok?

Generally, that's fine but in this case, it doesn't it's not sufficient. Why? Because the piling state is not reset. E.g., load the teapot example, pile by grid, then select flowers. The piles will still show teapots. While this also highlights another problem (the cover doesn't get updated as the textures get updated), the piles should not be kept between examples as it doesn't make sense. An easy fix is to give each set of drawings unique IDs. Maybe `${category}${i}`

**Pull request merged**